src/test/java/net/minecraftforge/test/WorldCapabilityRainTimerTest.java

```diff
@@ -0,0 +1,161 @@
+package net.minecraftforge.test;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.Maps;
+
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagIntArray;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.WorldInfo;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.Capability.IStorage;
+import net.minecraftforge.common.capabilities.CapabilityInject;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.ICapabilitySerializable;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+
+/** Simple mod to test world capability, by adding timer logic and check for raining time. */
+@Mod(modid=WorldCapabilityRainTimerTest.MODID, name="World Periodic Rain Check Test", version="0.0.0")
+public class WorldCapabilityRainTimerTest {
+    public static final String MODID = "worldperiodicrainchecktest";
+
+    @CapabilityInject(IRainTimer.class)
+    public static final Capability<IRainTimer> TIMER_CAP = null;
+
+    @EventHandler
+    public void init(FMLInitializationEvent event)
+    {
+        CapabilityManager.INSTANCE.register(IRainTimer.class, new TimerStorage(), DefaultTimer.class);
+        MinecraftForge.EVENT_BUS.register(new NormalEventHandler());
+    }
+
+    public static class NormalEventHandler {
+        @SubscribeEvent
+        public void attatchTimer(AttachCapabilitiesEvent.World event)
+        {
+            event.addCapability(new ResourceLocation(MODID, "rainTimer"), new RainTimerProvider());
+        }
+
+        @SubscribeEvent
+        public void onTick(TickEvent.WorldTickEvent event)
+        {
+            if(event.world.isRemote)
+                return;
+
+            IRainTimer timer = event.world.getCapability(TIMER_CAP, null);
+
+            if(event.phase == TickEvent.Phase.END)
+                timer.onTick();
+
+            if(event.phase == TickEvent.Phase.START){
+                if(!event.world.provider.getHasNoSky() && event.world.isRaining())
+                {
+                	if(timer.getDuration() == 0)
```

Comment:
tabs, and general code style in a few areas on this class, but not to important.